extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -100,17 +103,68 @@ export class AmpAdExit extends AMP.BaseElement {
       'CLICK_Y': true,
     };
     if (target.vars) {
-      for (const customVar in target.vars) {
-        if (customVar[0] == '_') {
-          vars[customVar] = () =>
-              args[customVar] || target.vars[customVar].defaultValue;
-          whitelist[customVar] = true;
+      const all3pResponses = this.getAmpDoc().win[AMP_ANALYTICS_3P_RESPONSES] ||
+          {};
+
+      for (const customVarName in target.vars) {
+        if (customVarName[0] == '_') {
+          const customVar =
+              /** @type {./config.Variable} */ (target.vars[customVarName]);
+          if (customVar) {
+            /*
+              Example:
+              The amp-ad-exit target has a variable representing the
+               priority of something, which is defined as follows:
+               "vars": {
+                 "_pty": {
+                   "defaultValue": "unknown",
+                   "vendorAnalyticsSource": "vendorXYZ",
+                   "vendorAnalyticsResponseKey": "priority"
+                 },
+                 ...
+               }
+               The cross-domain iframe of vendorXYZ has sent the
+               following response for the creative:
+                 { priority: medium, category: W }
+               This is just example data. The keys/values in that object can
+               be any strings.
+               The code below will create substitutionFunctions['_pty'],
+               which in this example will return "medium".
+             */
+            substitutionFunctions[customVarName] = () => {
+              if (customVar.hasOwnProperty('vendorAnalyticsSource') &&
+                  customVar.hasOwnProperty('vendorAnalyticsResponseKey')) {
+                // It's a 3p analytics variable
+                const vendor =
+                    /** @type {string} */ (customVar.vendorAnalyticsSource);
```

Comment:
I think we should use `customVar['vendorAnalyticsSource']` here, incase compiler rename this variable to something like `a.b`

Reply:
Good idea. Done.